stdlib/asyncio/coroutines.pyi

```diff
@@ -17,11 +17,15 @@ if sys.version_info < (3, 11):
     def coroutine(func: _FunctionT) -> _FunctionT: ...
 
 @overload
+@deprecated("Deprecated in Python 3.14; use inspect.iscoroutinefunction() instead")
 def iscoroutinefunction(func: Callable[..., Coroutine[Any, Any, Any]]) -> bool: ...
 @overload
+@deprecated("Deprecated in Python 3.14; use inspect.iscoroutinefunction() instead")
 def iscoroutinefunction(func: Callable[_P, Awaitable[_T]]) -> TypeGuard[Callable[_P, Coroutine[Any, Any, _T]]]: ...
 @overload
+@deprecated("Deprecated in Python 3.14; use inspect.iscoroutinefunction() instead")
 def iscoroutinefunction(func: Callable[_P, object]) -> TypeGuard[Callable[_P, Coroutine[Any, Any, Any]]]: ...
 @overload
+@deprecated("Deprecated in Python 3.14; use inspect.iscoroutinefunction() instead")
```

Comment:
For consideration, in https://github.com/python/typeshed/pull/14074 I'm also unconditionally marking these as deprecated, but @srittau mentionned that
> If I understand it correctly, it's sometimes necessary to use this on Python < 3.11, since it supports `@coroutine` (removed in 3.11), which the `inspect` version doesn't support.